lib/spack/spack/test/cmd/ci.py

```diff
@@ -1650,3 +1650,41 @@ def fake_dyn_mapping_urlopener(*args, **kwargs):
                 assert job.get("variables", {}).get("MY_VAR") == "hello"
                 assert "ignored_field" not in job
                 assert "unallowed_field" not in job
+
+
+def test_ci_generate_noop_no_concretize(
+    tmpdir,
+    working_env,
+    mutable_mock_env_path,
+    install_mockery,
+    mock_packages,
+    monkeypatch,
+    ci_base_environment,
+):
+    """This test assumes that the pkg-a package will never be edited."""
```

Comment:
Is this a safe assumption?  It was changed 2 months ago and 4 months ago.  I wonder if it's possible to instead define a package within the scope of the test somehow?

Reply:
I just saw this, maybe that works?

`repo.MockRepositoryBuilder(tmpdir)`

Reply:
Yeah, though I've never used it myself, that looks promising.

Reply:
duh, I am silly, I can mock `compute_affected_packages`